### ISSUE TITLE
Back-End: Create REST API Initial Skeleton

### ISSUE THREAD

**Issue by WMRamadan** [andersbandt/Financial-Analyzer#15]
Create a REST API to be used by the Front-End.

Tasks:

- [ ] Choose Framework
- [ ] Create Initial Structure
- [ ] Create Database Schema
- [ ] Create Initial End-Points
- [ ] Create Initial Tests

**Comment by WMRamadan** [comment]
@lordlabakdas So we are set for framework of choice being FastAPI?

**Comment by lordlabakdas** [comment]
We can go with fastapi if everyone agrees.

I think it really depends on the majority of us

**Comment by alex90271** [comment]
FastAPI would be good

**Comment by andersbandt** [comment]
All, please check out the poll at the Discord and vote on the framework of choice for the back end!

**Comment by nate-llc** [comment]
If it's the case that we're using flask, would flask-sqlalchemy be a good enough orm?

**Comment by lordlabakdas** [comment]
In my opinion, that would be best. Either flask-sqlalchemy or sqlalchemy. I
forget the difference.

**Comment by lordlabakdas** [comment]
sqlalchemy seems like a better bet. I see mixed reviews about flask-sqlalchemy.

**Comment by nate-llc** [comment]
The very interesting part of this project is the data scraping through monthly statements in statement_types/. The problem all of these classes have is that they inherit from Statement, which has several gui methods, and Statement inherits from Ledger which also is intertwined with gui methods and tkinter. I think it's going to be hard to build the db structure or anything related to the manipulation of data before we have written the extraction in a way that isn't coupled with gui functions

**Comment by lordlabakdas** [comment]
I agree @nate-llc. There needs to be decoupling of visualization and data analysis.

**Comment by YazidKurdi** [comment]
I suggest going top-down approach, rather than trying to fit the current intertwined mechanism of this project into a frontend/backend framework, we should:

1) Define the project from A-Z in written form (high level)
2) Division of frameworks (front/backend)
3) Within the frameworks division of sub-projects (Data load, data scrape, data viz, data base etc..)

This would be easier for each person to focus on a sub-project rather than each person focusing on everything.

**Comment by lordlabakdas** [comment]
@andersbandt 
This item on hold until above item 1 is addressed and an intial roadmap (task priorities) is provided.

**Comment by andersbandt** [comment]
@lordlabakdas agreed. I'm working on a write up now - might not be done until several hours from now with holiday stuff going on :)

**Comment by andersbandt** [comment]
Closing this issue as I have moved towards a CLI based application
